inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1319,6 +1319,69 @@ void GNAGraphCompiler::EltwisePrimitive(InferenceEngine::CNNLayerPtr layer) {
     }
 }
 
+void GNAGraphCompiler::GemmPrimitive(InferenceEngine::CNNLayerPtr layer) {
+    auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+
+    IE_ASSERT(!layer->insData.empty());
+    IE_ASSERT(!layer->outData.empty());
+    IE_ASSERT(layer->insData.size() == 2);
+    auto input_1 = layer->insData[0].lock();
+    auto input_2 = layer->insData[1].lock();
+    auto outputs = *layer->outData.begin();
+    auto inputPrecision = quantized ? Precision(Precision::I16) : input_1->getPrecision();
+    uint32_t noOfInputsDivisor = GNALimitations::noOfInputsDivisor;
+
+    auto input_data = HasTo2DReshapeData(layer) ? Get2DReshapedData(input_1, 8) : input_1;
+    auto in_dims = input_data->getDims();
+    auto batch_size = (in_dims.size() == 1) ? 1 : in_dims.front();
+    uint32_t num_rows_in = InferenceEngine::details::product(in_dims) / batch_size;
+    uint32_t num_columns_in = batch_size;
+    uint32_t num_rows_out = GetDataDimSize(outputs, 1);
+    uint32_t num_padding = ALIGN(num_rows_in, noOfInputsDivisor) - num_rows_in;
+    uint32_t num_padding_out = 0;
+
+    void* ptr_input_1 = nullptr;
+    void* ptr_outputs = nullptr;
+    void* ptr_input_2 = nullptr;
+    void* ptr_biases = nullptr;
+
+    auto& currentComponent = dnnComponents.addComponent(layer->name, ("affine"));
+
+    dnn->InitAffineComponent(currentComponent,
+                             num_rows_in + num_padding,
+                             num_columns_in,
+                             num_rows_out + num_padding_out,
+                             inputPrecision.size(),
+                             outputs->getPrecision().size(),
+                             quantized == nullptr ? input_2->getPrecision().size() : 2,
+                             quantized == nullptr ? input_2->getPrecision().size() : 4,
+                             quantized == nullptr ? 1 : quantized->_weights_quant.GetScale(),
+                             quantized == nullptr ? 1 : quantized->_dst_quant.GetScale(),
+                             ptr_input_1,
+                             ptr_outputs,
+                             ptr_input_2,
+                             ptr_biases,
+                             false);
+
+    size_t num_data_bytes_out = InferenceEngine::details::product(begin(outputs->getDims()), end(outputs->getDims()))
+                                * outputs->getPrecision().size();
+
+    size_t num_data_bytes_in_1 = InferenceEngine::details::product(begin(input_1->getDims()), end(input_1->getDims()))
+                               * input_1->getPrecision().size();
+    size_t num_data_bytes_in_2 = InferenceEngine::details::product(begin(input_2->getDims()), end(input_2->getDims()))
+                                 * input_2->getPrecision().size();
+
+    connectOutput(layer, ptr_outputs, num_data_bytes_out);
+    connectInput(layer, ptr_input_1, num_data_bytes_in_2);
```

Comment:
why is layer connected to ptr_input_1 with num_data_bytes_in_2 bytes? Doesn't it correspond to the first input?